crates/router/src/core/payment_methods/cards.rs

```diff
@@ -1275,28 +1276,29 @@ fn filter_pm_currencies_based(
     match (accepted_currency, req_currency_list) {
         (None, None) => (None, None, true),
         (None, Some(ref r)) => (
-            Some(admin::AcceptedCurrencies {
-                accept_type: "enable_only".to_owned(),
-                list: Some(r.to_vec()),
-            }),
+            Some(admin::AcceptedCurrencies::EnableOnly(r.to_vec())),
             Some(r.to_vec()),
             true,
         ),
         (Some(l), None) => (Some(l.to_owned()), None, true),
         (Some(l), Some(ref r)) => {
-            let list = if l.accept_type == "enable_only" {
-                filter_accepted_enum_based(&l.list, &Some(r.to_owned()))
-            } else {
-                filter_disabled_enum_based(&l.list, &Some(r.to_owned()))
+            let updated = match l {
+                admin::AcceptedCurrencies::EnableOnly(acc) => {
+                    filter_accepted_enum_based(&Some(acc.clone()), &Some(r.to_owned()))
+                        .map(admin::AcceptedCurrencies::EnableOnly)
+                }
+
+                admin::AcceptedCurrencies::DisableOnly(den) => {
+                    filter_disabled_enum_based(&Some(den.clone()), &Some(r.to_owned()))
+                        .map(admin::AcceptedCurrencies::DisableOnly)
+                }
```

Comment:
I'd prefer if the inner values were named `accepted` and `denied`, not a blocker though.

Reply:
That's what It was yesterday, `SpecificAccepted` and `SpecificDenied`. @Narayanbhat166 suggested keeping it as `EnableOnly` and `DisableOnly` to ensure the API contract stays constant which I felt was a valid suggestion.